File: DotNetAssignments/Loan/Loan/PersonalLoan.cs
namespace loanlib;

public class PersonalLoan : Loan, ITaxable 
{
     public override double Rate (int amount)
     {
        return amount <= 500000 ? 0.15 : 0.16;
     }

     public double CalculateTax(double amount)
     {
       return amount * 1.1;
     }
}

File: Java_master/JavaAssignment/LoanAssignment/loan/PersonalLoan.java
package loan;
public class PersonalLoan extends Loan implements Taxable{
    
    public PersonalLoan(double principle,int period){
        super(principle,period);
    }

    @Override
    public double getRate() {
        return super.getPrinciple()<=500000?0.15:0.16;
    }

    @Override
    public double getTax(double amount) {
       return amount*0.10;
    }

}
